AnnoMate/ReviewDataApp.py

```diff
@@ -375,7 +376,72 @@ def update_components(output_dict, subject_index_value, more_component_inputs_as
                 output_dict['annot_panel'] = current_annotations
             return output_dict
 
-        ####### Callbacks
+        ###### Callbacks
+        @app.callback(
+            output=dict(
+                is_open=Output('APP-freeze-modal', 'is_open')
+            ),
+            inputs=dict(
+                freeze_button=Input('APP-freeze-button', 'n_clicks'),
+                freeze_close=Input('APP-freeze-close', 'n_clicks'),
+                freeze_confirm=Input('APP-freeze-confirm', 'n_clicks')
+            ),
+            prevent_initial_call=True,
+        )
+        def toggle_freeze_modal(freeze_button, freeze_close, freeze_confirm):
+            if ctx.triggered_id=='APP-freeze-button':
+                return {'is_open': True}
+            return {'is_open': False}
+        
+        @app.callback(
+            output=dict(
+                subscript=Output('APP-title-subscript', 'children'),
+                subscript_color=Output('APP-title-subscript', 'style'),
+                button=Output('APP-freeze-button', 'children'),
+                history_display_table=Output('APP-history-table', 'data', allow_duplicate=True),
+                dropdown_options=Output('APP-dropdown-data-state', 'options', allow_duplicate=True),
+                annot_panel=annotation_panel_component.callback_output
+            ),
+            inputs=dict(
+                freeze_confirm=Input('APP-freeze-confirm', 'n_clicks'),
+                annotations_confirm=State('APP-annotations-checkbox', 'value'),
+                history_display_table=State('APP-history-table', 'data'),
+                dropdown_options=State('APP-dropdown-data-state', 'options'),
+                annot_panel=annotation_panel_component.callback_state 
+            ),
+            prevent_initial_call=True,
+        )
+        def freeze_data(freeze_confirm, annotations_confirm, history_display_table, dropdown_options, annot_panel):
+            subscript = 'TEST MODE  '
+            subscript_color = {'color': 'red'}
+            button = dbc.Button('Freeze Data', id='APP-freeze-button')
+
+            if freeze_confirm:
+                if not annotations_confirm:
+                    review_data.data.annot_df.loc[:,:]=None
```

Comment:
I think I would reset these variables more similar to how we initially instantiate them in `Data`

https://github.com/getzlab/AnnoMate/blob/7a2daf62120f3e4acb6dd5263a19d541e2abf703/AnnoMate/Data.py#L45-L55